src/legacy/core_plugins/vis_default_editor/public/components/controls/time_interval.tsx

```diff
@@ -56,15 +81,46 @@ function TimeIntervalParamEditor({
   let selectedOptions: ComboBoxOption[] = [];
   let definedOption: ComboBoxOption | undefined;
   let isValid = false;
+  let invalidEsMessage = '';
+  const errors = [];
   if (value) {
     definedOption = find(options, { key: value });
     selectedOptions = definedOption ? [definedOption] : [{ label: value, key: 'custom' }];
-    isValid = !!(definedOption || isValidInterval(value, timeBase));
+
+    // we check if Elasticsearch interval is valid ES interval to show a user appropriate error message
+    // we don't check if there is timeBase
+    const isValidIntervalValue = timeBase ? true : isValidEsInterval(value);
+
+    if (!isValidIntervalValue) {
+      invalidEsMessage = getInvalidEsMessage(value);
+    }
+
+    isValid = !!definedOption || (isValidIntervalValue && isValidInterval(value, timeBase));
+  }
+
+  let interval: { scaled: boolean; scale: number; expression: string } = {} as any;
+  if (isValid) {
+    interval = get(agg, 'buckets.getInterval') && (agg as any).buckets.getInterval();
+
+    // we check if Elasticsearch interval is valid to show a user appropriate error message
+    // e.g. there is the case when a user inputs '14d' but it's '2w' in expression equivalent and the request will fail
+    if (!invalidEsMessage && !isValidEsInterval(interval.expression)) {
+      isValid = false;
+      invalidEsMessage = getInvalidEsMessage(interval.expression);
+    }
+  }
+
+  if (!isValid) {
+    errors.push(
+      invalidEsMessage ||
+        i18n.translate('visDefaultEditor.controls.timeInterval.invalidFormatErrorMessage', {
+          defaultMessage: 'Invalid interval format.',
+        })
+    );
   }
 
-  const interval = get(agg, 'buckets.getInterval') && (agg as any).buckets.getInterval();
   const scaledHelpText =
-    interval && interval.scaled && isValid ? (
+    isValid && interval && interval.scaled ? (
```

Comment:
Does this inversion of sequence somehow affect final result?

Reply:
No. It's just more logical from my side to check if the input is valid first.